syncer/filter_test.go

```diff
@@ -50,82 +52,212 @@ func (s *testFilterSuite) TearDownSuite(c *C) {
 
 func (s *testFilterSuite) TestSkipQueryEvent(c *C) {
 	syncer := &Syncer{}
-	// build in skipDDL
-	skipped, err := syncer.skipQuery(nil, nil, "DROP TEMPORARY TABLE IF EXISTS `foo`.`bar` /* generated by server */")
+	// test binlog filter
+	filterRules := []*bf.BinlogEventRule{
+		{
+			SchemaPattern: "foo*",
+			TablePattern:  "",
+			Events:        []bf.EventType{bf.CreateTable},
+			SQLPattern:    []string{"^create\\s+table"},
+			Action:        bf.Ignore,
+		},
+	}
+	var err error
+	syncer.binlogFilter, err = bf.NewBinlogEvent(false, filterRules)
 	c.Assert(err, IsNil)
-	c.Assert(skipped, IsTrue)
 
-	// system table
-	skipped, err = syncer.skipQuery([]*filter.Table{{Schema: "mysql", Name: "test"}}, nil, "create table mysql.test (id int)")
+	cases := []struct {
+		sql           string
+		tables        []*filter.Table
+		expectSkipped bool
+	}{
+		{
+			// system table
+			"create table mysql.test (id int)",
+			[]*filter.Table{{Schema: "mysql", Name: "test"}},
+			true,
+		}, {
+			// test filter one event
+			"drop table foo.test",
+			[]*filter.Table{{Schema: "foo", Name: "test"}},
+			false,
+		}, {
+			"create table foo.test (id int)",
+			[]*filter.Table{{Schema: "foo", Name: "test"}},
+			true,
```

Comment:
maybe you can test RENAME in this case

Reply:
fixed in https://github.com/pingcap/dm/pull/2147/commits/4191f43ccd43e0027a5aab0d0bae64bc8ea7789e